src/input.rs

```diff
@@ -251,7 +251,7 @@ pub(crate) struct InputReader<'a> {
 impl<'a> InputReader<'a> {
     fn new<R: BufRead + 'a>(mut reader: R) -> InputReader<'a> {
         let mut first_line = vec![];
-        reader.read_until(b'\n', &mut first_line).ok();
+        Self::read_line_safely(&mut reader, &mut first_line).ok();
```

Comment:
We ignore the error here, but I don't think that is a problem in practice. Not ignoring the error here is likely to cascade into a larger change with higher risk.